spec/lib/log_parser_spec.rb

```diff
@@ -1,27 +1,43 @@
-require "log_parser"
+# frozen_string_literal: true
+
+require 'simplecov'
+SimpleCov.start
+require 'log_parser'
 
 describe LogParser do
-  describe "#get_first_line" do
-    it "gets first line" do
-      log_parser = LogParser.new("spec/fixtures/game_test.log")
-      first_line = log_parser.get_first_line
-      expect(first_line).to eq("  0:00 ------------------------------------------------------------")
+  describe '#first_line' do
+    context 'call first_line' do
+      it 'gets first line' do
+        log_parser = LogParser.new('spec/fixtures/game_test.log')
+        first_line = log_parser.first_line
+        expect(first_line).to eq('  0:00 ------------------------------------------------------------')
+      end
     end
   end
 
-  describe "#initialize" do
-    it "print file not found" do
-      expect { LogParser.new("game.log") }.to raise_error("File not found")
+  context 'when file does not exist' do
+    it 'print file not found' do
+      expect { LogParser.new('game.log') }.to raise_error('File not found')
     end
   end
-  
-  describe "#get_log_data do" do
-    it "prints lines number" do
-      log_parser = LogParser.new("spec/fixtures/game_test.log")
-      output = log_parser.get_log_data
-      expect(output).to eq("\"game_test.log\": {\n  \"lines\": 10\n}")
-      json_data = JSON.parse(output[17, output.length-1])
-      expect(json_data['lines']).to eq(10)
+
+  describe '#log_data' do
+    before do
+      log_parser = LogParser.new('spec/fixtures/game_test.log')
+      @output = log_parser.log_data
```

Comment:
Não é recomendado utilizar variáveis de instância no rspec para "facilitar" a comparação nos testes. Testes não devem ser "DRY" de uma lida nesse artigo:

https://thoughtbot.com/blog/four-phase-test

Reply:
Eita, essa alteração foi feita durante um pair com o Hallan. Ele que estava revisando minha tarefa 4, consequentemente atualizei a 3 para ficar da mesma forma.

Reply:
@grazirs o @brunohkbx não está errado, mas não explicou a questão por completo. Um passo por vez, veja a versão anterior do seu teste, demos somente um pequeno passo sem introduzir muitos novos conceitos. 

Mas vale uma explicação mais completa do que o @brunohkbx tá trazendo...

O `before` vai rodar antes de cada `it`, **SEMPRE**. Repetir o código a cada it também terá o mesmo efeito colateral, exatamente igual, ninguém ganha, ninguém perde. O real problema com as variáveis de instância é performático (não é a questão do DRY), uma vez que podemos simplesmente memoizá-las e ainda obtê-las de forma _lazy_. O rspec nos oferece o `subject` e o `let` pra esse fim. É importante notar que é muito valioso que em cada `describe/context` corresponda a `before/let/subject` reproduzindo o que o texto do `describe/context` descreve. Muitas vezes o `describe/context` aponta mudança no foco do teste (subject) ou parâmetros (lets e befores).

Segue um exemplo com subject:

```ruby
  describe '#log_data' do
    subject(:log_data) { LogParser.new('spec/fixtures/game_test.log').log_data }
    
    it 'prints file name' do
      is_expected_to.to start_with('"game_test.log":')
    end

```

Referências e boas práticas sobre subjects e lets: 
- https://www.betterspecs.org/#subject
- https://www.betterspecs.org/#let